build-system/tasks/cherry-pick.js

```diff
@@ -156,11 +156,13 @@ function cherryPick() {
       `Cherry-picked ${commits.length} commits onto release ${onto}`
     );
     process.exitCode = 0;
+    return Promise.resolve();
   } catch (e) {
     log(red('ERROR:'), e.message);
     log('Deleting branch', cyan(branch));
     getOutput(`git checkout master && git branch -d ${branch}`);
     process.exitCode = 1;
+    return Promise.reject();
```

Comment:
With promises, you shouldn't need `process.exitCode = 1`. Also, instead of rejecting without a reason, I think this code will work better. (The stack is hidden because it's mostly useless `nodejs` calls).

```js
    const rejectWithReason = (reasonText) => {
      const reason = new Error(reasonText);
      reason.showStack = false;
      return Promise.reject(reason);
    };
```

With this, you can replace every instance of `process.exitCode = 1` in the file with a call to `rejectWithReason('Some reason')`.

Reply:
Done